modules/unsupported/vector-mosaic/src/main/java/org/geotools/vectormosaic/VectorMosaicStore.java

```diff
@@ -0,0 +1,484 @@
+                                    typeName,
+                                    VectorMosaicGranule.fromDelegateFeature(delegateFeature));
+                    typeMap.put(typeName, typeConfiguration);
+                }
+            }
+        } catch (IOException e) {
+            LOGGER.log(
+                    Level.WARNING, "Failed to get feature source for type " + delegateTypeName, e);
+        } finally {
+            if (iterator != null) {
+                iterator.close();
+            }
+        }
+    }
+
+    /**
+     * Validates the schema of the delegate store.
+     *
+     * @param delegateStoreName the name of the delegate store.
+     * @param typeName the name of the type to validate.
+     * @param delegateSchema the schema of the type to validate.
+     * @return true if the schema is valid.
+     */
+    private boolean validateSchema(
+            String delegateStoreName, String typeName, SimpleFeatureType delegateSchema) {
+        for (String field : REQUIRED_VECTOR_MOSAIC_FIELDS.keySet()) {
+            if (delegateSchema.getDescriptor(field) == null) {
+                LOGGER.log(
+                        Level.INFO,
+                        "Delegate store "
+                                + delegateStoreName
+                                + " schema with type name "
+                                + typeName
+                                + " does not contain required field "
+                                + field);
+                return false;
+            } else if (!delegateSchema
+                    .getDescriptor(field)
+                    .getType()
+                    .getBinding()
+                    .getTypeName()
+                    .equals(REQUIRED_VECTOR_MOSAIC_FIELDS.get(field))) {
+                LOGGER.log(
+                        Level.INFO,
+                        "Delegate store "
+                                + delegateStoreName
+                                + " schema with type name "
+                                + typeName
+                                + " has the wrong field type for field "
+                                + field);
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Builds a qualified name from a name containing the ":" separator, otherwise the given name
+     * will be used as the local part
+     *
+     * @param name the name to build a qualified name from
+     * @return the qualified name
+     */
+    static Name buildName(String name) {
+        int idx = name.indexOf(":");
+        if (idx == -1) {
+            return new NameImpl(name);
+        } else {
+            String ns = name.substring(0, idx);
+            String local = name.substring(idx + 1);
+            return new NameImpl(ns, local);
+        }
+    }
+
+    @Override
+    protected List<Name> createTypeNames() throws IOException {
+        List<Name> result = new ArrayList<>();
```

Comment:
Here it should grab the delegate store and compute the type names list.

Reply:
Fixed